polly/src/pollable.rs

```diff
@@ -0,0 +1,197 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Portions Copyright 2017 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the THIRD-PARTY file.
+
+use std::fmt::Formatter;
+use std::os::unix::io::{AsRawFd, FromRawFd, RawFd};
+
+#[derive(Default, Clone, PartialEq, Debug)]
+pub struct Pollable {
+    // Event manager will use this in callbacks so the EventHandler implementation can
+    // multiplex the handling for multiple registered fds.
+    fd: RawFd,
+}
+
+/// Wrapper for file descriptorsl
+impl Pollable {
+    pub fn from<T: AsRawFd>(rawfd: &T) -> Pollable {
+        Pollable {
+            fd: rawfd.as_raw_fd(),
+        }
+    }
+}
+
+impl FromRawFd for Pollable {
+    unsafe fn from_raw_fd(fd: RawFd) -> Pollable {
+        Pollable { fd }
+    }
+}
+
+impl AsRawFd for Pollable {
+    fn as_raw_fd(&self) -> RawFd {
+        self.fd
+    }
+}
+
+pub type EventRegistrationData = (Pollable, EventSet);
+
+#[derive(PartialEq)]
+pub enum PollableOp {
+    /// Register a new handler for a pollable and eventset.
+    Register(EventRegistrationData),
+    /// Unregister a handler for a pollable.
+    Unregister(EventRegistrationData),
+    /// Update eventset for a specified pollable.
+    Update(EventRegistrationData),
+}
+
+impl std::fmt::Debug for PollableOp {
+    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
+        use self::PollableOp::*;
+
+        match self {
+            Register(data) => write!(f, "Register {:?}", data),
+            Unregister(data) => write!(f, "Unregister {:?}", data),
+            Update(data) => write!(f, "Update {:?}", data),
+        }
+    }
+}
+
+bitflags! {
+    #[derive(Default)]
+    pub struct EventSet: u8 {
+        const NONE = 0b0000_0000;
+        const READ = 0b0000_0001;
+        const WRITE = 0b0000_0010;
+        const CLOSE = 0b0000_0100;
+    }
+}
+
+impl EventSet {
+    /// Read event.
+    pub fn readable(self) -> bool {
```

Comment:
nit-ish: can we rename the functions to is_readable, is_writable... With this change, the code would read super nicely, almost like a regular English phrase:
```rust
        if event_type.is_readable() {
             ....
        }
```

Reply:
This is a good idea.